nova_plugin/server.py

```diff
@@ -261,25 +261,22 @@ def create(nova_client, neutron_client, args, **kwargs):
 
     if external_server:
         _set_network_and_ip_runtime_properties(external_server)
-        if ctx._local:
+        network_ids = \
+            get_openstack_ids_of_connected_nodes_by_openstack_type(
+                ctx, NETWORK_OPENSTACK_TYPE)
+        port_ids = get_openstack_ids_of_connected_nodes_by_openstack_type(
+            ctx, PORT_OPENSTACK_TYPE)
+        try:
+            _validate_external_server_nics(
+                external_server,
+                network_ids,
+                port_ids
+            )
+            _validate_external_server_keypair(nova_client)
             return
-        else:
-            network_ids = \
-                get_openstack_ids_of_connected_nodes_by_openstack_type(
-                    ctx, NETWORK_OPENSTACK_TYPE)
-            port_ids = get_openstack_ids_of_connected_nodes_by_openstack_type(
-                ctx, PORT_OPENSTACK_TYPE)
-            try:
-                _validate_external_server_nics(
-                    neutron_client,
-                    network_ids,
-                    port_ids
-                )
-                _validate_external_server_keypair(nova_client)
-                return
-            except Exception:
-                delete_runtime_properties(ctx, RUNTIME_PROPERTIES_KEYS)
-                raise
+        except Exception:
```

Comment:
What exception are you trying to catch here? let's make sure we are not using blanket excepting catching.

Reply:
Yes, nothing critical can't raise here, and we don't save anything to runtime properties, only recheck and attach ports to vm, so we can skip cleanup.